asdf/core/_converters/integer.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+
+from asdf import constants
+from asdf.extension import Converter
+from asdf.tags.core import IntegerType
+
+
+class IntegerConverter(Converter):
+    tags = [
+        "tag:stsci.edu:asdf/core/integer-1.0.0",
+        "tag:stsci.edu:asdf/core/integer-1.1.0",
+    ]
+
+    types = [
```

Comment:
Related to https://github.com/asdf-format/asdf/pull/1474#discussion_r1138969192

I wonder if this PR and #1474 might benefit from a more flexible way of matching types to converters. I have not looked at this recently (so please correct any mistakes  I make as I'm typing from memory) but I believe the types are only used during conversion from a custom to tagged tree (for writing or validation). If the goal of this converter is (in part) to support all 'inty' types from numpy, inspecting those at runtime should be fine (given that we only need to support the objects that are supported by the system running this code).

One other comment and thing to discuss is that this doesn't round trip types. For example:
```python
t = {'a': numpy.uint8(255), 'b': numpy.uint8(255)}
rt = asdf.testing.helpers.roundtrip_object(tree)
rt['a'] + rt['b'] == 510
t['a'] + t['b'] == 254   # and a RuntimeWarning: overflow encountered in scalar add
```